File: packages/sdk-ruby/lib/bidstack_crm/resources/contacts.rb
require_relative "base"

module BidstackCrm
  module Resources
    class Contacts < Base
      def list(page: 1, per_page: 50, q: nil)
        params = { page: page, perPage: per_page }
        params[:q] = q if q
        get("/contacts", params: params)
      end

      def find(contact_id) = get("/contacts/#{contact_id}")
      def create(data, idempotency_key: nil) = post("/contacts", body: data, idempotency_key: idempotency_key)
      def update(contact_id, data) = patch("/contacts/#{contact_id}", body: data)
      def destroy(contact_id) = delete("/contacts/#{contact_id}")
    end
  end
end

File: packages/sdk-python/bidstack_crm/resources/contacts.py
"""Contacts resource."""
from __future__ import annotations
from typing import Any
from ._base import BaseResource


class ContactsResource(BaseResource):
    def list(self, *, page: int = 1, per_page: int = 50, q: str | None = None) -> dict[str, Any]:
        params: dict[str, Any] = {"page": page, "perPage": per_page}
        if q:
            params["q"] = q
        return self._get("/contacts", params=params)

    def get(self, contact_id: str) -> dict[str, Any]:
        return self._get(f"/contacts/{contact_id}")

    def create(self, data: dict[str, Any], idempotency_key: str | None = None) -> dict[str, Any]:
        return self._post("/contacts", json=data, idempotency_key=idempotency_key)

    def update(self, contact_id: str, data: dict[str, Any]) -> dict[str, Any]:
        return self._patch(f"/contacts/{contact_id}", json=data)

    def delete(self, contact_id: str) -> None:
        self._delete(f"/contacts/{contact_id}")
